File: algorithms/0347.top-k-frequent-elements/solution.go
func topKFrequent(nums []int, k int) []int {
	m := map[int]int{}
	for _, num := range nums {
		m[num]++
	}

	type count struct {
		Key   int
		Value int
	}

	var numsCount []count
	for k, v := range m {
		numsCount = append(numsCount, count{k, v})
	}
	sort.Slice(numsCount, func(i, j int) bool {
		return numsCount[i].Value > numsCount[j].Value
	})

	res := []int{}
	for _, x := range numsCount {
		if len(res) < k {
			res = append(res, x.Key)
		} else {
			break
		}
	}

	return res
}

File: algorithms/0347.top-k-frequent-elements/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn top_k_frequent(nums: Vec<i32>, k: i32) -> Vec<i32> {
        let mut count: HashMap<i32, i32> = HashMap::new();
        for n in nums {
            let entry = count.entry(n).or_insert(0);
            *entry += 1;
        }

        let mut count_vec: Vec<(&i32, &i32)> = count.iter().collect();
        count_vec.sort_unstable_by(|a, b| b.1.cmp(a.1));
        count_vec.iter().map(|x| *(x.0)).take(k as usize).collect()
    }
}
